### DISCUSSION TITLE
Can I use h2 schema.sql to initialize my database without using flyway?

### DISCUSSION THREAD

**Discussion by Manfred73** [quarkusio/quarkus#30193] [Community]
I have migrated my spring service to Quarkus, but running into an issue with some integration tests.

One of my integration tests checks an error if a duplicate record is inserted into the database (unique constraint set on one of the fields in a db table).

In spring I used h2 and schema.sql to create the db schema. This schema.sql also created the (unique) indexes and the tables. And for the repository test I used @DataJpaTest(properties = "spring.jpa.generate-ddl=true") .

Now in Quarkus, the schema is created based on the entities (and no indexes are created), thus my integration test for checking duplicate fails.

Is there a way to tell quarkus it should use the schema.sql under src/main/resources to create the database instead of using the entities for that?
Is there also an alternative for spring.jpa.generate-ddl=true?

**Comment by arey** [comment]
Hi @Manfred73
Did you find a solution to use your `schema.sql` script file?
Something is missing from my Quarkus configuration:
```
%test.quarkus.hibernate-orm.database.generation=none
%test.quarkus.hibernate-orm.sql-load-script=db/h2/data.sql
```

**Reply by OndraZizka** [reply]
If I understand you correctly, what you could do is using the datasource config.

The extension passes some of the variables to the JDBC URL as `;key=val;...`

And H2 takes the `INIT` param where you can execute a script.
https://www.h2database.com/html/features.html#execute_sql_on_connection

IDK how to pass them though. In the source it is declared as 

```
    /**
     * Generic properties that are passed for additional container configuration.
     * <p>
     * Properties defined here are database-specific
     * and are interpreted specifically in each database dev service implementation.
     */
    @ConfigDocMapKey("property-key")
    Map<String, String> containerProperties();

    /**
     * Generic properties that are added to the database connection URL.
     */
    @ConfigDocMapKey("property-key")
    Map<String, String> properties();
```

**Reply by OndraZizka** [reply]
Note that despite being named containerProperties, I tracked it to be applied to the URL. Maybe that's a bug in H2 impl and the `properties` should have been used.
